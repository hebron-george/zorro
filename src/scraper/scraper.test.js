var fs = require('fs');

import {
    loadNamesCSV,
    scrape,
    findPersonID,
    getHtmlFileName,
    saveHtml,
    getPoliticoHtml,
    getResumeLink,
} from './scraper';

import htmlPolitico from './html/politico.html.js';
import htmlFichaTab1 from './html/ficha-tab1.html.js';

describe('scraper', () => {
    describe('loadNamesCSV', () => {
        it('should load the csv', () => {
            return loadNamesCSV(`${__dirname}/names.test.csv`)
                .then((result) => {
                    expect(result).toMatchSnapshot();
                });
        });
    });

    describe('findPersonID', () => {
        it('find the person id in the html', () => {
            expect(findPersonID(htmlPolitico)).toMatchSnapshot();
        });
    });

    describe('getHtmlFileName', () => {
        it('formats the html file name', () => {
            const fileType = 'resume';
            const person = {
                apellido_materno: 'ALVARADO',
                apellido_paterno: 'ABAD',
                nombres: 'THALIA',
                rowid: 1,
            };
            const personId = 666;
            expect(getHtmlFileName(fileType, person, personId)).toMatchSnapshot();
        });
    });

    describe('saveHtml', () => {
        it('creates a file and doesnt blow up', () => {
            const fileType = 'resume';
            const person = {
                apellido_materno: 'ALVARADO',
                apellido_paterno: 'ABAD',
                nombres: 'THALIA',
                rowid: 1,
            };
            const personId = 666;
            return saveHtml(fileType, person, personId, 'asdf');
        });
    });

    describe('getPoliticoHtml', () => {
        xit('should actually make a request and return the html', () => {
            const person = {
                apellido_materno: 'ALVARADO',
                apellido_paterno: 'ABAD',
                nombres: 'THALIA',
                rowid: 1,
            };

            return getPoliticoHtml(person)
                .then((html) => {
                    expect(html).toContain('ficha.aspx');
                })
        })
    });

    describe('getResumeLink', () => {
        it('should find the resume link for 2014', () => {
            expect(getResumeLink(htmlFichaTab1)).toMatchSnapshot();
        })
    });
});
