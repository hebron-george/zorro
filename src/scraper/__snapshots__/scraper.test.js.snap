exports[`scraper findPersonID find the person id in the html 1`] = `16097875`;

exports[`scraper getHtmlFileName formats the html file name 1`] = `"resume-666-THALIA-ABAD-ALVARADO.html"`;

exports[`scraper getResumeLink should find the resume link for 2014 1`] = `"http://aplicaciones007.jne.gob.pe/pecaoe/sipe/HojaVida.aspx?cod=u80uW4sKN%2fGGkRNlqSDOQw%3d%3d"`;

exports[`scraper loadNamesCSV should load the csv 1`] = `
Array [
  Object {
    "apellido_materno": "ALVARADO",
    "apellido_paterno": "ABAD",
    "nombres": "THALIA",
    "rowid": 1
  },
  Object {
    "apellido_materno": "CASIMIRO",
    "apellido_paterno": "ABAD",
    "nombres": "DARWIN EUSTORGIO",
    "rowid": 2
  }
]
`;
